backend/gateway/mux/mux.go

```diff
@@ -171,9 +171,9 @@ func customResponseForwarder(ctx context.Context, w http.ResponseWriter, resp pr
 		http.SetCookie(w, cookie)
 	}
 
-	if redirects := md.HeaderMD.Get("Location"); len(redirects) > 0 {
-		w.Header().Set("Location", redirects[0])
-
+	// Redirect if it's the browser (non-XHR).
+	redirects := md.HeaderMD.Get("Location")
+	if len(redirects) > 0 && isBrowser(copyRequestHeadersFromResponseWriter(w)) {
```

Comment:
super nit: I think this might be easier to read as `requestHeadersFromResponseWriter`